amip_interp/amip_interp.F90

```diff
@@ -146,18 +146,17 @@ module amip_interp_mod
 !-----------------------------------------------------------------------
 !----------------- Public interfaces -----------------------------------
 
-public amip_interp_init, get_amip_sst, get_amip_ice, amip_interp_new, &
-       amip_interp_del, amip_interp_type, assignment(=)
+public amip_interp_init, amip_interp_init_r4, amip_interp_init_r8, get_amip_sst, &
+     & get_amip_ice, amip_interp_new, amip_interp_del, amip_interp_type, assignment(=)
 
 !-----------------------------------------------------------------------
 !----------------- Public Data -----------------------------------
 integer :: i_sst = 1200
 integer :: j_sst = 600
-real, parameter:: big_number = 1.E30
+real(r8_kind), parameter:: big_number = 1.E30
 logical :: forecast_mode = .false.
-real, allocatable, dimension(:,:) ::  sst_ncep, sst_anom
-
-public i_sst, j_sst, sst_ncep, sst_anom, forecast_mode, use_ncep_sst
```

Comment:
not sure about changing the public variable names or making some of these private  in mixedmode.  Perhaps these changes could be proposed to main instead.

Reply:
One alternative could be to create `sst_ncep` and `sst_anom` pointers as aliases for the r8 versions, but this seems unnecessary if no model code is actually using them.

Reply:
They could be used by non-GFDL models, @bensonr @laurenchilutti?

Reply:
@mlee03 @laurenchilutti - those public entities from amip_interp_mod are used in [external_sst.F90](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/main/tools/external_sst.F90) in all branches and [GFDL/atmosphere.F90](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/blob/main/driver/GFDL/atmosphere.F90) for GFDL climate models.

Reply:
@bensonr thank you, @J-Lentz would you be able to change these back to their original names? :)

Reply:
Sorry about that, I'd gotten the impression that this module is only used by `ice_amip` and `ocean_amip`. d8d70837fea3f26e79da605ff45ca0abcfd34515 restores the public data and adds public pointers to the r8 versions of `sst_ncep` and `sst_anom`.